src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -312,7 +312,11 @@ public int compare(TypedVar o1, TypedVar o2) {
       Set<String> propertyNames =
           objType != null ? objType.getPropertyNames() : Collections.<String>emptySet();
       for (String property : propertyNames) {
-        if (!isPrivateProperty(objType, property)) {
+        // When parsing externs namespaces are explicitly declared with a var of Object type
+        // Do not emit the var declaration, as it will conflict with the namespace.
+        JSType nativeObjectType = compiler.getTypeRegistry().getNativeType(OBJECT_TYPE);
+        if (!isPrivateProperty(objType, property) &&
+            !(isExtern && isLikelyNamespace(objType.getPropertyType(property), nativeObjectType))) {
```

Comment:
given all the `!`s, would it make sense to invert either the conditions or the whole if statement? DeMorganify it?